tests/b2share_unit_tests/files/test_b2share_storage.py

```diff
@@ -214,3 +216,121 @@ def test_embargoed_records_with_external_pids(app, test_users,
                                        key=file.obj.key)
                 resp = client.get(ext_file_url)
                 assert resp.status_code == 302
+
+
+def test_generation_external_pids(app, records_data_with_external_pids,
+                                  deposit_with_external_pids):
+    """Test the generate_external_pids function."""
+    expected_output = records_data_with_external_pids['external_pids']
+    with app.app_context():
+        deposit = Deposit.get_record(deposit_with_external_pids.deposit_id)
+        output = generate_external_pids(deposit)
+        assert output == expected_output
+
+        first_file = list(deposit.files)[0]
+        first_file.delete(first_file.obj.bucket, first_file.obj.key)
+        output = generate_external_pids(deposit)
+        # new_expected_output = expected_output[1:]
+        assert_external_files(deposit, output)
+
+
+def test_getting_record_with_external_pids(app, login_user, test_users,
+                                           deposit_with_external_pids,
+                                           records_data_with_external_pids):
+    """External pids are retrieved properly when getting a record."""
+
+    def test_get_deposit(deposit_pid_value, user):
```

Comment:
Note for later: We should create some helper function to simplify that. It is done in other tests too I think.